lib/rubocop/cop/rails/action_filter.rb

```diff
@@ -74,16 +77,22 @@ def message(node)
         end
 
         def bad_methods
+          return ACTION_METHODS if filter_only?
           style == :action ? FILTER_METHODS : ACTION_METHODS
         end
 
         def good_methods
+          return FILTER_METHODS if filter_only?
           style == :action ? ACTION_METHODS : FILTER_METHODS
         end
 
         def preferred_method(method)
           good_methods[bad_methods.index(method.to_sym)]
         end
+
+        def filter_only?
```

Comment:
I guess you could have overriden the `style` method or simply disabled the cop on Rails 3 (which was actually my initial idea).